client/landing/stepper/declarative-flow/internals/steps-repository/unified-domains/index.tsx

```diff
@@ -93,6 +98,20 @@ export default function DomainsStep( props: StepProps ) {
 		},
 		[ stepState, setStepState ]
 	);
+	useEffect( () => {
+		// This will automatically submit the domains step when navigating back from checkout
+		// It will redirect the user to plans step. The idea is to achieve the same behavior as /start.
+		const signupDestinationCookieExists = retrieveSignupDestination();
+		const isReEnteringFlow = getSignupCompleteFlowName() === props.flow;
+		const isManageSiteFlow = Boolean(
+			wasSignupCheckoutPageUnloaded() && signupDestinationCookieExists && isReEnteringFlow
+		);
+		if ( isManageSiteFlow ) {
+			if ( Object.keys( stepState ).length > 0 ) {
```

Comment:
I think this check is always true.
At minimum the step name will be always there